Maquette 2.4.X Regressions

**Bug**

It appears that Maquette 2.4.X has introduced some regressions in our code that was unexpected.  Upon compiling Widgets now we get the following errors:

```
Using tsc v2.0.3
src/mixins/createFormFieldMixin.ts(112,6): error TS2450: Left-hand side of assignment expression cannot be a constant or a read-only property.
src/mixins/createFormFieldMixin.ts(115,5): error TS2450: Left-hand side of assignment expression cannot be a constant or a read-only property.
src/mixins/createFormFieldMixin.ts(117,6): error TS2450: Left-hand side of assignment expression cannot be a constant or a read-only property.
src/mixins/createFormFieldMixin.ts(120,5): error TS2450: Left-hand side of assignment expression cannot be a constant or a read-only property.
src/mixins/createRenderMixin.ts(230,6): error TS2450: Left-hand side of assignment expression cannot be a constant or a read-only property.
src/mixins/createRenderMixin.ts(231,6): error TS2450: Left-hand side of assignment expression cannot be a constant or a read-only property.
src/mixins/createRenderMixin.ts(232,6): error TS2450: Left-hand side of assignment expression cannot be a constant or a read-only property.
src/projector.ts(154,5): error TS2450: Left-hand side of assignment expression cannot be a constant or a read-only property.
src/projector.ts(160,4): error TS2450: Left-hand side of assignment expression cannot be a constant or a read-only property.
src/projector.ts(161,4): error TS2450: Left-hand side of assignment expression cannot be a constant or a read-only property.
```

The functional tests also appear to fail, but it is hard to tell if it is related to these complication errors or further breaking changes in Maquette.

Reverting to `maquette@2.3.7` appears to alleviate the problem.

Package Version: 2.0.0-alpha

**Code**

```
> npm install maquette@latest
```

**Expected behavior:**

It works, without issue...

**Actual behavior:**

It has issues.

--- a/src/mixins/createFormFieldMixin.ts
+++ b/src/mixins/createFormFieldMixin.ts
@@ -106,20 +106,10 @@ const createFormMixin: FormMixinFactory = compose({
 
 		nodeAttributes: [
 			function (this: FormFieldMixin<any, FormFieldMixinState<any>>): VNodeProperties {
-				const props: VNodeProperties = {};
+				const { type, value, state } = this;
+				const { disabled, name } = state;
 
-				if (this.type) {
-					props['type'] = this.type;
-				}
-				/* value should always be copied */
-				props.value = this.value;
-				if (this.state && this.state.name) {
-					props.name = this.state.name;
-				}
-
-				props.disabled = Boolean(this.state.disabled);
-
-				return props;
+				return { type, value, name, disabled: Boolean(disabled) };
 			}
 		]
 	}, (instance: FormField<any>, { type } = <any> {}) => {

--- a/src/mixins/createRenderMixin.ts
+++ b/src/mixins/createRenderMixin.ts
@@ -212,10 +212,8 @@ const createRenderMixin = createStateful
 
 			nodeAttributes: [
 				function (this: RenderMixin<RenderMixinState>): VNodeProperties {
-					const props: VNodeProperties = this.state && this.state.id
-						? { 'data-widget-id': this.state.id }
-						: {};
-
+					const baseIdProp = this.state && this.state.id ? { 'data-widget-id': this.state.id } : {};
+					const styles = this.styles || {};
 					const classes: { [index: string]: boolean; } = {};
 					const widgetClasses = widgetClassesMap.get(this);
 
@@ -226,10 +224,7 @@ const createRenderMixin = createStateful
 						widgetClassesMap.set(this, this.classes);
 					}
 
-					props.classes = classes;
-					props.styles = this.styles || {};
-					props.key = this;
-					return props;
+					return Object.assign(baseIdProp, { key: this, classes, styles });
 				}
 			],
 

--- a/src/mixins/createRenderMixin.ts
+++ b/src/mixins/createRenderMixin.ts
@@ -224,7 +224,7 @@ const createRenderMixin = createStateful
 						widgetClassesMap.set(this, this.classes);
 					}
 
-					return Object.assign(baseIdProp, { key: this, classes, styles });
+					return assign(baseIdProp, { key: this, classes, styles });
 				}
 			],
 

--- a/src/projector.ts
+++ b/src/projector.ts
@@ -146,10 +146,23 @@ const noopHandle = { destroy() { } };
 const emptyVNode = h('div');
 const noopVNode = function(): VNode { return emptyVNode; };
 
+interface ProjectorAttributes {
+
+	classes?: {
+		[index: string]: boolean | null | undefined;
+	};
+
+	styles?: {
+		[index: string]: string | null | undefined;
+	};
+
+	[index: string]: any;
+}
+
 export const createProjector: ProjectorFactory = compose<ProjectorMixin, ProjectorOptions>({
 		getNodeAttributes(this: Projector, overrides?: VNodeProperties): VNodeProperties {
 			/* TODO: This is the same logic as createCachedRenderMixin, merge somehow */
-			const props: VNodeProperties = {};
+			const props: ProjectorAttributes  = {};
 			for (let key in this.listeners) {
 				props[key] = this.listeners[key];
 			}

